src/Symfony/Component/Security/Http/RememberMe/RememberMeDetails.php

```diff
@@ -21,14 +21,12 @@ class RememberMeDetails
 {
     public const COOKIE_DELIMITER = ':';
 
-    private $userFqcn;
     private $userIdentifier;
     private $expires;
     private $value;
 
-    public function __construct(string $userFqcn, string $userIdentifier, int $expires, string $value)
+    public function __construct(string $userIdentifier, int $expires, string $value)
```

Comment:
This is a BC break.

Reply:
See e.g. https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/Security/Http/Firewall/AccessListener.php on how to remove parameters in a BC matter

Reply:
You mean that I have to use trigger_deprecation ?

Reply:
Yes, but the most important bit is that both `new RememberMeDetails(User::class, $userId, $expiry, $value);` and `new RememberMeDetails($userId, $expiry, $value);` must be valid and work.

This means you have to remove the type from all arguments (as you want to remove the first one). We "dummy" type them (e.g. `__construct(/* string */ $userIdentifier, /* int */ $expires, ...)`).

Then, check which signature is used (e.g. check if `$expires` is an int. If it is, the new signature is used, if it isn't the old one is used. Then, map the variables to the correct name accordingly.

Another real world example: https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/Security/Core/Authorization/AuthorizationChecker.php#L35

Reply:
That's way harder than excepted! Challenge accepted :)

Reply:
I'm not sure about how \func_num_args() > 4 ? func_get_arg(4) : 'value?'; should work here..

Reply:
The more I code, the less I get about how it could work .. I mean, $this->userFqcn would be useless. But in your last example, the "removed" authenticationManager is still used. :/

Reply:
I get that func_num_args should help to set default value but there's no default value in this construct :/

Reply:
I have trouble following these example but I may be on a good path. I close this in favor of new reviews on newly pushed code 🙏